includes/classes/SiteAutomation/EndPoints.php

```diff
@@ -0,0 +1,343 @@
+			'sanitize_callback' => 'sanitize_text_field',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		/**
+		 * Filters the get sophi posts query params.
+		 *
+		 * @param array $params Query params.
+		 *
+		 * @since 1.3.0
+		 */
+		return apply_filters( 'site_automation_params', $params );
+	}
+
+	/**
+	 * Update the sophi posts.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @param  WP_REST_Request $request The request.
+	 * @return mixed
+	 */
+	public function site_automation_override( $request ) {
+		$current_user = wp_get_current_user();
+
+		if ( $current_user->exists() ) {
+			$user_email = $current_user->user_email;
+		} else {
+			return new \WP_Error( 401, __( 'Unauthorised user, please log in.', 'sophi-wp' ) );
+		}
+
+		// Get the auth token.
+		$api_token = $this->auth->get_access_token();
+
+		if ( is_wp_error( $api_token ) ) {
+			return new \WP_Error( 401, __( 'Invalid API token, please try adding correct credentials on the settings page.', 'sophi-wp' ) );
+		}
+
+		// Request parameters.
+		$timeout    = 3;
+		$attributes = $request->get_query_params();
+
+		$api_url = get_sophi_settings( 'sophi_override_url' );
+		$host    = get_sophi_settings( 'host' );
+		$api_url = $api_url . 'hosts/' . $host . '/overrides';
+
+		$rule_type        = isset( $attributes['ruleType'] ) ? sanitize_title( $attributes['ruleType'] ) : '';
+		$page_name        = isset( $attributes['pageName'] ) ? sanitize_title( $attributes['pageName'] ) : '';
+		$override_post_ID = isset( $attributes['overridePostID'] ) ? sanitize_title( $attributes['overridePostID'] ) : '';
+		$widget_name      = isset( $attributes['widgetName'] ) ? sanitize_title( $attributes['widgetName'] ) : '';
+		$override_expiry  = isset( $attributes['overrideExpiry'] ) ? sanitize_title( $attributes['overrideExpiry'] ) : 2;
+		$position         = isset( $attributes['position'] ) ? sanitize_title( $attributes['position'] ) : 1;
+
+		if ( empty( $widget_name ) && 'ban' !== $rule_type ) {
+			return new \WP_Error( 401, __( 'Missing parameter: widgetName', 'sophi-wp' ) );
+		}
+
+		$body = array(
+			"articleId"           => $override_post_ID,
+			"expirationHourOfDay" => $override_expiry,
+			"page"                => $page_name,
+			"position"            => 'ban' === $rule_type || 'out' === $rule_type ? '' : $position,
+			"requestedUserName"   => $user_email,
+			"ruleType"            => 'ban' === $rule_type ? 'out' : $rule_type,
+			"widgetName"          => 'ban' === $rule_type ? null : $widget_name,
+		);
+		$body = wp_json_encode( $body );
+		$args = [
+			'method'  => 'POST',
+			'headers' => array(
+				'Content-Type'  => 'application/json',
+				'Authorization' => 'Bearer ' . $api_token,
+				'Cache-Control' => 'no-cache',
+			),
+			'body'    => $body,
+		];
+
+		/**
+		 * Filters the arguments being passed to the override api auth request.
+		 *
+		 * @param array $args Arguments.
+		 * @param string $api_url Auth API URL.
+		 *
+		 * @since 1.3.0
```

Comment:
Same comment as the other filter, a few changes needed here for our hookdocs

Reply:
https://github.com/globeandmail/sophi-for-wordpress/pull/333/commits/d7f3bb5f76704d21c9cf317832c74ef2d0842ca3